rust/src/http2/http2.rs

```diff
@@ -441,6 +474,9 @@ pub struct HTTP2State {
     dynamic_headers_tc: HTTP2DynTable,
     transactions: VecDeque<HTTP2Transaction>,
     progress: HTTP2ConnectionState,
+    // layered packets contents for DNS over HTTP2
+    layered: Vec<Vec<u8>>,
+    dns_state: Option<*mut std::os::raw::c_void>,
```

Comment:
TODO comment

I am not sure if this should be generic or DNS-specific...

Reply:
Is DoH a little special in that it doesn't follow the normal PROTO-over-HTTP formats like something like GRPC might do?  Where they just use the bodies for their payload?

Reply:
> Where they just use the bodies for their payload?

Indeed DoH is not exactly like that : it has DNS message in the URI for requests (a base64-encoded parameter), and in bodies for responses.

Why does it matter ?

Reply:
> > Where they just use the bodies for their payload?
> 
> Indeed DoH is not exactly like that : it has DNS message in the URI for requests (a base64-encoded parameter), and in bodies for responses.
> 
> Why does it matter ?

Say we were to support gRPC over HTTP, and Avro over HTTP, and Thrift over HTTP. Those would also make sense to have something generic.  As DoH is different, could it be put within the same generic container or does it always need some special handling due to be in the URI.  I guess not. But that's the path of thought I was on.

Reply:
Indeed I was hesitating for naming `dns_state` or something more generic...

Reply:
I'm OK without over thinking making it more generic until we have the need to make it more generic.